soda_machine.rb

```diff
@@ -7,12 +9,25 @@ def initialize(args = {})
   end
 
   def current_inventory_count
-  end
+   @sodas.count
+ end
 
-  def find_soda(soda_brand)
+ def find_soda(soda_brand)
+  @sodas.each_with_index do |soda, index|
+    if soda.brand == soda_brand
```

Comment:
If you're only returning a value, you could write this as `@sodas[index] if soda.brand == soda_brand`